### ISSUE TITLE
Optional "client name" option in connection

### ISSUE THREAD

**Issue by samueleaton** [jgaskins/nats#1]
from the nats doc, we can specify an optional connection name https://docs.nats.io/reference/reference-protocols/nats-protocol#syntax-1

i went through the code in this repo and found out this is achievable by appending a path to the connection url.

**EXAMPLE**
![Screen Shot 2022-03-15 at 5 06 24 PM](https://user-images.githubusercontent.com/9119269/158492167-93abba2f-2ddb-489a-98b9-d17236c4b3be.png)
![Screen Shot 2022-03-15 at 5 06 39 PM](https://user-images.githubusercontent.com/9119269/158492183-03f1c0b9-1e73-4e89-8818-a3c2a17f4a47.png)

I'm very new to NATS, so am wondering if its common in NATS clients to append the client name to the connection string. If not, I wonder if that would be better as an optional arg to the Client `initialize` method, then it would be auto-documented in the API docs.

**Comment by samueleaton** [comment]
https://docs.nats.io/using-nats/developer/connecting/name

looks like other clients take it as an optional parameter

**Comment by jgaskins** [comment]
Yeah, I think what I was going for here was similar to the Postgres-style URI where the path points to the DB name. I wouldn't mind adding support for a separate argument, though.

**Comment by samueleaton** [comment]
@jgaskins fyi I have a [PR on this](https://github.com/samueleaton/nats/pull/1) that's based on my other one. Will point it here once the other one is merged. here: 

I made it so you can give it a name option but it will fallback to the URI path. The benefit of the URI path is that you could give different names to different servers if connecting to multiple (not sure if useful).
